gpustack/worker/inference_backend_manager.py

```diff
@@ -119,3 +122,20 @@ def _handle_event(self, event: Event):
 
         except Exception as e:
             logger.error(f"Error handling InferenceBackend event: {e}")
+
+    async def check_docker_hub_reachable(self):
+        """
+        Check if Docker Hub is reachable.
+        To avoid frequent checks, cache the result for a short period via global lock.
+
+        Returns:
+            bool: True if Docker Hub is reachable, False otherwise.
+        """
+        try:
+            resp = requests.get("https://registry-1.docker.io/v2/", timeout=3)
+            _is_docker_hub_reachable = resp.status_code < 500
+            logger.debug(f"Docker Hub reachable: {_is_docker_hub_reachable}")
+        except Exception as e:
+            _is_docker_hub_reachable = False
+            logger.warning(f"Docker Hub connectivity check failed: {e}")
```

Comment:
log the error in debug. Use info to tell users Docker Hub connectivity check failed and we are gonna switch to use quay.io